src/components/InputNumber/index.tsx

```diff
@@ -41,16 +41,41 @@ const InputNumber: FC<IInputNumber> = ({
       if (isNaN(numericValue)) {
         return null;
       }
+      const minValueString = minValue.toString();
+      if (minValueString.length > value.length) {
+        return numericValue;
+      }
```

Comment:
Here at this point you can simply compare if the value is lower, no more need to check the length of the strings since the check on minimum value is done when `blur` event occurs